lib/fabrik/DirectorManager.js

```diff
@@ -240,6 +245,30 @@ class DirectorManager extends BaseManager {
       });
   }
 
+  executePreUpdateActions(deploymentName, context) {
```

Comment:
can you rename the function to `executePreUpdate` ? by dropping the actions suffix, want to consciously keep this separate from the deployment hooks a.k.a. actions. Also this should have been just `executePreUpdateAction` in the first place as its just one action :-)